app/src/main/java/org/mozilla/fenix/compose/list/ListItem.kt

```diff
@@ -0,0 +1,342 @@
+    iconDescription: String? = null,
+    onClick: (() -> Unit)? = null,
+) {
+    Box(
+        modifier = when (onClick != null) {
+            true -> Modifier.clickable { onClick() }
+            false -> Modifier
+        }.then(
+            Modifier
+                .padding(horizontal = 16.dp)
+                .defaultMinSize(minHeight = LIST_ITEM_HEIGHT),
+        ),
+        contentAlignment = Alignment.Center,
+    ) {
+        Icon(
+            painter = painterResource(icon),
+            contentDescription = iconDescription,
+            modifier = Modifier.size(ICON_SIZE),
+            tint = FirefoxTheme.colors.iconPrimary,
+        )
+    }
+}
+
+@Composable
+@Preview(name = "TextListItem", uiMode = Configuration.UI_MODE_NIGHT_YES)
+private fun TextOnlyListItemPreview() {
+    FirefoxTheme {
+        Box(Modifier.background(FirefoxTheme.colors.layer1)) {
+            TextListItem(label = "Label only")
+        }
+    }
+}
+
+@Composable
+@Preview(name = "TextListItem with a description", uiMode = Configuration.UI_MODE_NIGHT_YES)
+private fun ListItemDescriptionPreview() {
+    FirefoxTheme {
+        Box(Modifier.background(FirefoxTheme.colors.layer1)) {
+            TextListItem(
+                label = "Label + description",
+                description = "Description text"
+            )
+        }
+    }
+}
+
+@Composable
+@Preview(name = "TextListItem with an icon", uiMode = Configuration.UI_MODE_NIGHT_YES)
+private fun ListItemRightIconPreview() {
+    FirefoxTheme {
+        Box(Modifier.background(FirefoxTheme.colors.layer1)) {
+            TextListItem(
+                label = "Label + right icon",
+                icon = R.drawable.ic_menu,
+                iconDescription = "click me",
+            )
+        }
+    }
+}
+
+@Composable
+@Preview(name = "DrawableListItem", uiMode = Configuration.UI_MODE_NIGHT_YES)
+private fun ListItemLeftIconPreview() {
+    FirefoxTheme {
+        Box(Modifier.background(FirefoxTheme.colors.layer1)) {
+            DrawableListItem(
+                label = "Label + left icon",
+                drawable = R.drawable.ic_folder_icon,
+                drawableDescription = "click me",
+            )
+        }
+    }
+}
+
+@Composable
+@Preview(name = "DrawableListItem with an interactable icon", uiMode = Configuration.UI_MODE_NIGHT_YES)
+private fun ListItemMultiIconPreview() {
+    FirefoxTheme {
+        Box(Modifier.background(FirefoxTheme.colors.layer1)) {
+            DrawableListItem(
+                label = "Label + left/right icons",
+                drawable = R.drawable.ic_folder_icon,
+                drawableDescription = null,
+                icon = R.drawable.ic_menu,
+                iconDescription = "click me",
+                onIconClick = { println("icon click") },
+            )
+        }
+    }
+}
+
+@Composable
+@Preview(name = "FavIconListItem with an icon and onClicks", uiMode = Configuration.UI_MODE_NIGHT_YES)
+private fun CompleteListItemPreview() {
```

Comment:
Any particular reason these shouldn't just be named after their component `FaviconListItemPreview`? I feel the the @Preview name does provide a helpful description of the intended preview so I am not quite sure if the actual function name needs to be different from the component. I would typically look for the 1:1 component to preview naming if I was trying to see an example of the @Preview

Reply:
The function name is leftover from V1, so that's my bad for not updating it after giving the custom `@Preview` name. I think that's a fair assessment though. For this, I was testing how it looked using a custom preview name to compare apples to apples, but I agree that the function name should align with the component itself.